code_analysis.py

```
@@ -107,33 +120,66 @@ def __init__(self, project_path: str):
+        
+        try:
+            # Load sentence transformer model
+            self.embedding_model = SentenceTransformer('all-MiniLM-L6-v2')
+            
+            # Initialize FAISS index
+            dimension = 384  # Dimension for all-MiniLM-L6-v2
+            self.vector_db = faiss.IndexFlatL2(dimension)
+            
+            logger.info("Vector embeddings initialized for semantic search")
+            
+        except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)